addons/account/models/account_move.py

```diff
@@ -3342,13 +3342,13 @@ def _reverse_moves(self, default_values_list=None, cancel=False):
                 skip_invoice_sync=move.move_type == 'entry',
             ).copy(default_values)
 
-        reverse_moves.with_context(skip_invoice_sync=cancel).write({'line_ids': [
+        reverse_moves.write({'line_ids': [
             Command.update(line.id, {
                 'balance': -line.balance,
                 'amount_currency': -line.amount_currency,
             })
             for line in reverse_moves.line_ids
-            if line.move_id.move_type == 'entry' or line.display_type == 'cogs'
+            if line.move_id.move_type == 'entry' or line.display_type in ('cogs', 'epd')
```

Comment:
Instead of copying the incorrect epd lines, and then inverting them, could we rather copy the payment term when necessary, and allow the epd lines to be computed?

Here's what I'm thinking (maybe with shorter lines 😄 ) WDYT?

```py
    def _prepare_default_reversal(self, move):
        reverse_date = self.date if self.date_mode == 'custom' else move.date
        mixed_payment_term = move.invoice_payment_term_id.id if move.invoice_payment_term_id and move.company_id.early_pay_discount_computation == 'mixed' else None
        return {
            'ref': _('Reversal of: %(move_name)s, %(reason)s', move_name=move.name, reason=self.reason)
                   if self.reason
                   else _('Reversal of: %s', move.name),
            'date': reverse_date,
            'invoice_date_due': reverse_date,
            'invoice_date': move.is_invoice(include_receipts=True) and (self.date or move.date) or False,
            'journal_id': self.journal_id.id,
            'invoice_payment_term_id': None,
            'invoice_user_id': move.invoice_user_id.id,
            'auto_post': 'at_date' if reverse_date > fields.Date.context_today(self) else 'no',
            'invoice_payment_term_id': mixed_payment_term,
        }
```

Reply:
Thanks for the review and the suggestion, it is working as expected and it's much more readable